compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -364,6 +365,12 @@ private class ExtractDependenciesCollector extends tpd.TreeTraverser { thisTreeT
           addImported(sel.name)
           if sel.rename != sel.name then
             addUsedName(sel.rename, UseScope.Default)
+      case exp @ Export(expr, selectors) =>
+        val dep = expr.tpe.classSymbol
+        if dep.exists && selectors.exists(_.isWildcard) then
```

Comment:
A comment explaining why we need to be careful with wildcards would be useful since it's a bit subtle.

Reply:
added comment below